### ISSUE TITLE
Typed Properties and default column/association types

### ISSUE THREAD

**Issue by beberlei** [doctrine/orm#7939]
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes
| RFC         | no
| BC Break    | no

#### Summary

When declaring a type on a property, this should act as a default for field type or ManyToOne/OneToOne targetEntity.

```php
class User
{
    /** @ORM\Column */
    public int $id; // defaults to type="integer"

   /** @ORM\ManyToOne */
   public Email $email; // defaults to targetEntity=Email::class
}
```

The relevant code would be in `ClassMetadataInfo::mapField`, `mapManyToOne` and `mapOneToOne`. Whenever `type` or `targetEntity` is missing, it could look this up in reflection property instead if it has a type.

**Comment by renanbr** [comment]
Could we expect it'll also detect `nullable` properties?

```php
/** @ORM\Column */
public ?string $name;
```

**Comment by Lustmored** [comment]
I have submitted #8439 as an attempt to implement features discussed in this issue. It probably still needs work, but it would be great to get some feedback :)

**Comment by beberlei** [comment]
Fixed in #8439
